File: app/src/main/java/com/apt/delegate/source/delegate/RemoteDelegate.java
package com.apt.delegate.source.delegate;

import com.apt.delegate.source.IRemoteDataSource;

public class RemoteDelegate implements IRemoteDataSource {

    @Override
    public void getRemoteUser() {
        //do something
    }

    @Override
    public void getRemoteUserList() {
        //do something
    }

    @Override
    public void getRemoteBook() {
        //do something
    }

    @Override
    public Object getOtherData() {
        //do something
        return null;
    }
}

File: app/src/main/java/com/apt/delegate/source/delegate/RemoteDelegate.kt
package com.apt.delegate.source.delegate

import com.apt.delegate.source.IRemoteDataSource

class RemoteDelegate : IRemoteDataSource {

    override fun getRemoteUser() {
        //do something
    }

    override fun getRemoteUserList() {
        //do something
    }

    override fun getRemoteBook() {
        //do something
    }

    override fun getOtherData(): Any? {
        //do something
        return null
    }
}
